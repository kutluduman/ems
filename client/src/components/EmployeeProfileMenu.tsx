import { alpha, styled } from "@mui/material/styles";
import Popover from "@mui/material/Popover";
import { Theme } from "@mui/material";

const ArrowStyle = styled("span")(({ theme }) => ({
  [theme.breakpoints.up("sm")]: {
    top: -7,
    zIndex: 1,
    width: 12,
    right: 20,
    height: 12,
    content: "''",
    position: "absolute",
    borderRadius: "0 0 4px 0",
    transform: "rotate(-135deg)",
    background: theme.palette.background.paper,
    borderRight: `solid 1px ${alpha(theme.palette.grey[500], 0.12)}`,
    borderBottom: `solid 1px ${alpha(theme.palette.grey[500], 0.12)}`,
  },
}));

interface EmployeeProfileMenuProps {
  [key: string]: any;
}

const EmployeeProfileMenu = ({
  children,
  sx,
  open,
  ...other
}: EmployeeProfileMenuProps) => {
  return (
    <Popover
      anchorOrigin={{ vertical: "bottom", horizontal: "right" }}
      transformOrigin={{ vertical: "top", horizontal: "right" }}
      open={open}
      PaperProps={{
        sx: {
          mt: 1.5,
          ml: 0.5,
          overflow: "inherit",
          border: (theme: Theme) => `solid 1px ${theme.palette.grey[500]}`,
          width: 200,
          ...sx,
        },
      }}
      {...other}
    >
      <ArrowStyle className="arrow" />

      {children}
    </Popover>
  );
}
export default EmployeeProfileMenu;