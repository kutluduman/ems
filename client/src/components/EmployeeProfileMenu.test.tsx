import { create } from "react-test-renderer";

import EmployeeProfileMenu from "./EmployeeProfileMenu";

describe("EmployeeProfileMenu Component", () => {
  test("renders the EmployeeProfileMenu Component", () => {
    const tree = create(<EmployeeProfileMenu open={false} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});