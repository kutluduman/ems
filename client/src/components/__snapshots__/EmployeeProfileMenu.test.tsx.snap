// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeProfileMenu Component renders the EmployeeProfileMenu Component 1`] = `null`;
